### DISCUSSION TITLE
Decremental While Loop

### DISCUSSION THREAD

**Discussion by GoodluckH** [tqdm/tqdm#1164] [Q&A] [answered]
I was wondering if there's anyway to implement tqdm on a while-loop that terminates at certain minimal value rather than an upper bound? I don't quite know how to rewrite this as a for-loop to use tqdm. 

For example:
```python
while h > 0:
    ....
    h /= 3
```

Thank you.

**Comment by casperdcl** [comment, marked as answer]
```python
while h > 0:
    h /= 3
```

will technically be an infinite loop if it weren't for machine precision.

You could do:

```python
with tqdm() as t:
    while h > 0:
        h /= 3
        t.update()
```

or if instead you mean:

```python
while h > 1:
    h /= 3
```

then you could do:

```python
import math
from tqdm import trange

total = int(math.ceil(math.log(h, 3)))
for _ in trange(total):
    if h <= 1:
        break
    h /= 3
```

**Reply by GoodluckH** [reply]
Thank you, this is very helpful!
